### ISSUE TITLE
Publish code training for Hawpv3

### ISSUE THREAD

**Issue by aquexexl** [cherubicXN/hawp#43]
Hi,
I 've a question, have you got a date when the HAWPv3 training code will be publish ? Otherwide, how can I train Hawp on my own dataset ? I've read the topic about "training own dataset" but I didn't interstand it

Thank you !

**Comment by cherubicXN** [comment]
Hi, I have uploaded the training code. Please give me some time to fill in the description of how to train.

**Comment by aquexexl** [comment]
Nice!
You are awesome !
Have you got some advises (some parameters to modify ) to train image with no perspective like ortho-images  ?

**Comment by cherubicXN** [comment]
For the ortho-images, did you mean the satellite images? 
Usually, I do not tune the hyperparameters as long as the loss items look normal. 
For my hawpv3, I did not train or tune it on the satellite images yet.

**Comment by aquexexl** [comment]
Hi, no I mean ortho-images like this : https://www.caroline-thibault-architecte.com/photos/orthophoto-ouestencorereduite.jpg

**Comment by cherubicXN** [comment]
Because we did not set any assumptions for our method, I think the default setting would be feasible. If you want to train your own model, the training log of the line segment recall would be an important cue to see if the network hits the line segments well.

**Comment by aquexexl** [comment]
Thank you for your advise. Have you got the file for hawp/ssl/datasets/ in order to use JSON from labelme (like this example)  ?                        
 {
  "version": "5.1.0",
  "flags": {},
  "shapes": [
    {
      "label": "toiture",
      "points": [
        [
          2901.7543859649118,
          8759.561403508773
        ],
        [
          2863.1578947368416,
          9271.842105263158
        ]
      ],
      "group_id": null,
      "shape_type": "line",
      "flags": {}
    }

**Comment by aquexexl** [comment]
Hi !
Forget my last question, I tried to use your train code for HAWPv3 but I get some problemz with the dataset. I can use wireframe dataset, your loader use a yaml file unlike HAWPv2. Is it possible to have a modified code to load wireframe dataset like HAWPv2 ?
Thank you in advance, I really apreciate your work !

**Comment by cherubicXN** [comment]
Hi @aquexexl, sorry for the very late response to your issue. I think the code modification should be easy as long as you can load the annotated data into the dataset. Maybe you need to write a new dataset class to achieve this.
